api/users/users.js

```diff
@@ -105,6 +122,35 @@ router.post('/remove_recipes', asyncHandler(async (request, response, next) => {
 
   user = await user.save();
   response.sendStatus(200);
+
+  /* can perform after response, send events to recommendation engine */
+  const userHash = crypto.createHash('sha256').update(request.openid.user.sub).digest('hex');
+  const removals = [];
+  recipes.forEach((recipeId) => {
+    removals.push(new rqs.DeletePurchase(userHash, recipeId, { cascadeCreate: true }));
+  });
+
+  await recombeeClient.send(new rqs.Batch(removals)).catch((err) => {
+    if (err) next(err);
+  });
+}));
+
+router.post('/recently_viewed', asyncHandler(async (request, response, next) => {
+  const {
+    count = 10,
+  } = request.body;
+
+  const userHash = crypto.createHash('sha256').update(request.openid.user.sub).digest('hex');
+  await recombeeClient.send(
+    new rqs.RecommendItemsToUser(userHash, count, { scenario: 'recently_viewed' }),
+  )
+    .then((recommendation) => {
```

Comment:
Does this mean that on the frontend you have to send when a user viewes a recipe directly TO recombee, or is there an endpoint through the server for that?

Reply:
There is an endpoint, it's right above this one